bigquery2/jwt_issuer.bal

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/encoding;
+import ballerina/internal;
+import ballerina/crypto;
+
+# Issue a JWT token.
+#
+# + header - JwtHeader object
+# + payload - JwtPayload object
+# + config - JWTIssuerConfig object
+#
+# + return - JWT token string or an error if token validation fails
+public function issue(internal:JwtHeader header, internal:JwtPayload payload, internal:JWTIssuerConfig config)
+                                                                                       returns (string|error) {
+    string jwtHeader = check createHeader(header);
+    string jwtPayload = check createPayload(payload);
+    string jwtAssertion = jwtHeader + "." + jwtPayload;
+
+    crypto:KeyStore keyStore = {
+        path: config.keyStoreFilePath,
+        password: config.keyStorePassword
+    };
+    var privateKey = check crypto:decodePrivateKey(keyStore = keyStore, keyAlias = config.keyAlias,
+        keyPassword = config.keyPassword);
+    string signature = "";
+    if (header.alg == "RS256") {
+        signature = encodeBase64Url(check crypto:signRsaSha256(jwtAssertion.toByteArray("UTF-8"), privateKey));
+    } else if (header.alg == "RS384") {
+        signature = encodeBase64Url(check crypto:signRsaSha384(jwtAssertion.toByteArray("UTF-8"), privateKey));
+    } else if (header.alg == "RS512") {
+        signature = encodeBase64Url(check crypto:signRsaSha512(jwtAssertion.toByteArray("UTF-8"), privateKey));
+    } else {
+        error jwtError = error(AUTH_ERROR_CODE, { message: "Unsupported JWS algorithm" });
+        return jwtError;
+    }
+
+    return (jwtAssertion + "." + signature);
+}
+
+# Construct JWT Header.
+#
+# + header - JwtHeader object
+#
+# + return - Encoded JWT Header or an error if token validation fails
+function createHeader(internal:JwtHeader header) returns (string|error) {
+    json headerJson = {};
+    if (!validateMandatoryJwtHeaderFields(header)) {
+        error jwtError = error(INTERNAL_ERROR_CODE, { message: "Mandatory field signing algorithm(alg) is empty." });
```

Comment:
It is incorrect/misleading to use a Ballerina stdlib module name in the error reason. What's the need for that?

Reply:
Same comment goes for other instances where this is used